graphlearn_torch/csrc/cpu/random_sampler.cc

```diff
@@ -71,6 +74,62 @@ CPURandomSampler::SampleWithEdge(const torch::Tensor& nodes, int32_t req_num) {
   return std::make_tuple(nbrs, nbrs_num, out_eid);
 }
 
+std::tuple<torch::Tensor, torch::Tensor>
+CPURandomSampler::WeightedSample(const torch::Tensor& nodes, int32_t req_num) {
```

Comment:
It's better to move this weightedsample in a separate file.

Reply:
Done!